api/swagger.yaml

```diff
@@ -0,0 +1,788 @@
+      properties:
+        name:
+          type: string
+        password:
+          type: string
+        player:
+          type: string
+    Tick:
+      type: object
+      properties:
+        tick:
+          type: integer
+          description: Game tick in milliseconds
+          format: int64
+          example: 10
+        wait:
+          type: boolean
+          default: true
+
+    Coordinates:
+      type: object
+      properties:
+        x:
+          type: integer
+          format: int64
+        y:
+          type: integer
+          format: int64
+      example:
+        x: 10
+        y: 20
+
+    StaticData:
+      type: object
+      properties:
+        classes:
+          type: object
+          additionalProperties:
+            type: object
+            properties:
+              name:
+                type: string
+              shipyard:
+                type: boolean
+              speed:
+                type: integer
+                format: int64
+              cargo:
+                type: integer
+                format: int64
+              life:
+                type: integer
+                format: int64
+              damage:
+                type: integer
+                format: int64
+        resources:
+          type: object
+          additionalProperties:
+            type: string
+      example:
+        classes: {
+          "1": {
+            name: "mothership",
+            shipyard: true,
+            speed: 10,
+            cargo: 100,
+            life: 360,
+            damage: 30,
+          },
+          "2": {
+            name: "hauler",
+            shipyard: false,
+            speed: 13,
+            cargo: 200,
+            life: 200,
+            damage: 0,
+            price: 500000
+          }
+        }
+        resources: {
+         "1": "spice melange",
+         "2": "pokeballs"
+        }
+              
+    GameObject:
+      required:
+      - id
+      - type
+      type: object
+      properties:
+        type:
+          type: string
+        id:
+          type: integer
+          format: int64
+      example:
+        id: 0
+        type: type
+      # discriminator:
+      #   propertyName: Type_
+      # oneOf:
+      # - $ref: '#/components/schemas/Ship'
+      # - $ref: '#/components/schemas/MyShip'
+    CargoItem:
+      type: object
+      properties:
+        resource:
+          type: integer
+          format: int64
+        amount:
+          type: integer
+          format: int64
+    Color:
+      type: object
+      properties:
+        r:
+          type: integer
+          format: int64
+        g:
+          type: integer
+          format: int64
+        b:
+          type: integer
+          format: int64
+    Game:
+      type: object
+      properties:
+        season:
+          type: integer
+          format: int64
+        tick:
+          type: integer
+          format: int64
+    Planet:
+      type: object
+      properties:
```

Comment:
planet can also have cargo/resources/commodities. including one or both of _sellprice_ and _buyprice_

Reply:
Should it really have only oneof sellprice or buyprice? IMHO it makes sense to have both - some players will sell their goods there, while others will buy them (for a bigger price than it was sold of course). How else are resources going to appear on planets?

Reply:
it may have both.
planets function as assemblers in factorio, they transform resources to other resources.

Reply:
Sorry I've read your comment before as a one of instead of one or